File: src/main/java/com/bible/week03/examples/TopLevel.java
package com.bible.week03.examples;

public class TopLevel {
    public static void main(String[] args) {
        // getter/setter로 접근
        String separator = Constants.getUNIX_LINE_SEPARATOR();

        // 변경 가능한 프로퍼티
        Constants.setDebugMode(true);
        boolean debug = Constants.getDebugMode();

        // 계산된 프로퍼티
        String systemSep = Constants.getCurrentSystemLineSeparator();
    }
}

File: src/main/kotlin/week03/examples/TopLevel.kt
@file:JvmName("Constants")
package com.bible.week03.examples

// Top-level properties
val UNIX_LINE_SEPARATOR = "\n"
val WINDOWS_LINE_SEPARATOR = "\r\n"

// 계산된 프로퍼티
val currentSystemLineSeparator: String
    get() = System.getProperty("line.separator")

// 변경 가능한 프로퍼티
var debugMode = false

// 사용 예
fun printWithSystemSeparator(text: String) {
    print(text + currentSystemLineSeparator)
}

fun main() {
    debugMode = true
    println("Debug mode: $debugMode")

    printWithSystemSeparator("Hello")
    printWithSystemSeparator("World")

    //StringUtils.joinToString<String>(listOf<String>("a", "b"), "", "")
}